lib/orocos/name_service.rb

```diff
@@ -650,7 +645,16 @@ def deregister(name)
             #(see NameServiceBase#validate)
             def validate
                 CORBA.refine_exceptions("corba naming service #{ip}") do
-                    do_validate
+                    begin
+                        do_validate
+                    rescue Orocos::ComError => e
+                        CORBA.warn "Name service is unreachable: #{e.message}\n"
+                        puts "You can try to fix this manually by restarting the nameserver:"
```

Comment:
Why not use `CORBA.warn` here as well ? It would at least provide a mean to silence the message if one would want to.

Reply:
Good point!